client/src/components/ResultPanel/ResultPanel.ts

```diff
@@ -62,5 +51,56 @@ export const showResult = (html: string, uri?: Uri, title?: string) => {
       true,
     );
   }
+
+  html = html
+    // Inject vscode context into our results html body
+    .replace(
+      "<body ",
+      `<body data-vscode-context='${JSON.stringify({
+        preventDefaultContextMenuItems: true,
+        resultPanelId,
+      })}' `,
+    )
+    // Make sure the html and body take up the full height of the parent
+    // iframe so that the context menu is clickable anywhere on the page
+    .replace(
+      "</head>",
+      `<script language="javascript">
+          if(acquireVsCodeApi){
+            const vscode = acquireVsCodeApi();
+            const panelId = '${resultPanelId}'
+            vscode.setState({panelId});
+          }
+         </script>
+         <style>html,body { height: 100% !important; }</style></head>`,
+    );
   resultPanel.webview.html = html;
+  setContextValue(resultPanelId, html);
```

Comment:
There's a problem. When "SAS.results.singlePanel" set to `true`.
First time say you have `resultPanelId === '001'`, it will create a result panel with
`resultPanel.onDidDispose(() => disposePanel('001'));`
Next time you have `resultPanelId === '002'`. So in your context, you have
```
{
    '001': 'html content 1',
    '002': 'html content 2',
}
```
But when the panel closes, it will only call `disposePanel('001')`. Then you have stale content left in the context.
I would suggest the `resultPanelId` should really match the "panel" instead of "content". When a panel is re-used, we should use the original panel id instead of generating a new id.